tools/common/StaticRegistrar.cs

```diff
@@ -5234,6 +5234,95 @@ public void Register (PlatformResolver resolver, IEnumerable<AssemblyDefinition>
 			}
 		}
 
+		static bool IsPropertyTrimmed (PropertyDefinition pd, AnnotationStore annotations)
+		{
+			if (pd is null)
+				return false;
+
+			if (!IsTrimmed (pd, annotations))
+				return false;
+			if (pd.GetMethod is not null && !IsTrimmed (pd.GetMethod, annotations))
+				return false;
+			if (pd.SetMethod is not null && !IsTrimmed (pd.SetMethod, annotations))
+				return false;
+			return true;
+		}
+
+		public static bool IsTrimmed (MemberReference tr, AnnotationStore annotations)
+		{
+			if (tr is null)
+				return false;
+
+			var assembly = tr.Module?.Assembly;
+			if (assembly is null) {
+				// Trimmed away
+				return true;
+			}
+
+			var action = annotations.GetAction (assembly);
+			switch (action) {
+			case AssemblyAction.Skip:
+			case AssemblyAction.Copy:
+			case AssemblyAction.CopyUsed:
+			case AssemblyAction.Save:
+				return false;
+			case AssemblyAction.Link:
+				break;
+			case AssemblyAction.Delete:
+				return true;
+			case AssemblyAction.AddBypassNGen:
+			case AssemblyAction.AddBypassNGenUsed:
+			default:
+				throw ErrorHelper.CreateError (99, $"Unknown linker action: {action}");
+			}
+
+			if (annotations.IsMarked (tr))
+				return false;
+			
+			if (annotations.IsMarked (tr.Resolve ()))
+				return false;
+
+			return true;
+		}
+
+		public void FilterTrimmedApi (AnnotationStore annotations)
+		{
+			var trimmedAway = Types.Where (kvp => IsTrimmed (kvp.Value.Type, annotations)).ToArray ();
```

Comment:
Are the `ToArray` calls for making debugging easier? Because I don't see how you need them here.

Reply:
No, it's because I'm removing items from the `Types` dictionary two lines below, and if I'm also enumerating over the dictionary, an exception will be thrown (it's not allowed to enumerate over a dictionary while the dictionary is being modified).

The ToArray call makes sure we're no enumerating over the `Types` dictionary below.